numpy/core/src/multiarray/flagsobject.c

```diff
@@ -207,15 +208,36 @@ arrayflags_dealloc(PyArrayFlagsObject *self)
         return item; \
     }
 
+static char *msg = "future versions will not create a writeable "
+    "array from broadcast_array. Set the writable flag explicitly to "
+    "avoid this warning.";
+
+#define _define_get_warn(UPPER, lower) \
+    static PyObject * \
+    arrayflags_ ## lower ## _get(PyArrayFlagsObject *self) \
+    { \
+        PyObject *item; \
+        if (self->flags & NPY_ARRAY_WARN_ON_WRITE) { \
+            if (PyErr_Warn(PyExc_FutureWarning, msg) < 0) {\
+                return NULL; \
+            } \
+        }\
+        item = ((self->flags & (UPPER)) == (UPPER)) ? Py_True : Py_False; \
+        Py_INCREF(item); \
+        return item; \
```

Comment:
nit: could combine these three lines and the declaration to `return PyBool_FromLong((self->flags & (UPPER)) == (UPPER))` (same on line 206). But I suppose this is consistent with what was there before

Reply:
adopting